### ISSUE TITLE
[Bug]: accessing the deferred thumbnail URL doesn't add it to the thumbnail cache

### ISSUE THREAD

**Issue by dkarlovi** [pimcore/pimcore#15233]
### Pimcore version

10.6

### Steps to reproduce

We've discussed this before in https://github.com/pimcore/pimcore/issues/12055 and other related issues / PRs.

Basically, this situation will never fix itself:

1. asset last modified M
2. thumbnail file exists on storage, last modified M + 1
3. `assets_image_thumbnail_cache` entry doesn't exist

In this case, the thumbnail URL will always be the deferred one. When I access it, it will read from storage as is:

https://github.com/pimcore/pimcore/blob/ec3e8a2abe625f2ade45ee1b62cd79947b90f58f/models/Asset/Image/Thumbnail/Processor.php#L196-L200

but it will never add it to the cache. This means the cache will be broken until we rerun `pimcore:thumbnail:image` which explicitly fixes this case.

### Expected / actual behavior

Accessing deferred thumbnail URL is the worst case scenario because you want it to trigger as little as possible. If it's accessed, the cache entry for the thumbnail **must** be up to date. It being in cache also enables it to serve the CDN version of the URL.

Note: this was quite difficult to debug because of #15231 which has the same symptom, but caused by totally different cause.

**Comment by dvesh3** [comment]
@dkarlovi could you please confirm if this is fixed with https://github.com/pimcore/pimcore/pull/15323? thanks!

**Comment by dkarlovi** [comment]
@dvesh3 these are separate issues, I wouldn't expect both to be solved by that fix.

**Comment by dvesh3** [comment]
@dkarlovi thing is...I'm not able to reproduce it. could you please share reproducible steps?

**Comment by dkarlovi** [comment]
Will do in 30mins.

**Comment by dkarlovi** [comment]
@dvesh3 

OK, here's the reproducer:

1. add a new image asset, for example ID `$x=6453`
2. fetch a path of a thumbnail for it with
```php
$image = Asset::getById(6453);
$thumb = $image->getThumbnail('coreshop_productDetail');
$path = $thumb->getPath(); // will be something like /_default_upload_bucket/6453/image-thumb__6453__coreshop_productDetail/poster.jpeg, the "deferred thumbnail URL"
```
3. make sure there isn't an entry for the asset in thumbnail cache (simulating a corrupted cache entry, an alternative would be to modify the `modificationDate` field to be older than the asset)
```sql
DELETE FROM assets_image_thumbnail_cache WHERE cid = 6453;
```
4. **visit** the URL in question in the browser, it gets served by Pimcore
5. check the thumbnail cache:
```sql
SELECT COUNT(*) FROM assets_image_thumbnail_cache WHERE cid = 6453;
```

At this point, the thumbnail should be in cache again to ensure the next time `getPath()` is called, it will produce the "non-deferred path (which is then used by Pimcore when generating the `<picture/>` element). But, it isn't, it will forever serve the deferred URL which is then handled by Pimcore over and over and over, the most expensive serving of images possible.

The bug here is the cache will never correct itself on its own, you MUST rerun the thumbnail generator command which only recently fixes this too, we specifically needed to fix that.

Basically, ["Pimcore serving thumbnails" should be seen as the absolutely worst case possible](https://github.com/pimcore/pimcore/issues/15231#issuecomment-1572012136) and we should ensure it happens the least possible number of times. If Pimcore served the thumbnail once, it means it needed to do a lot of work, let's not let that work be redone.

**Comment by dkarlovi** [comment]
@dvesh3 is this enough info to reproduce, does it make sense? :thinking:

**Comment by dvesh3** [comment]
Sorry, I didn't get a chance to reproduce it and will do it next week. thanks for the details!

**Comment by brusch** [comment]
Solution is probably to refresh the thumbnail cache here: 
https://github.com/pimcore/pimcore/blob/be8eb3faa9d5e2b0517574acdb9fd6de5157e04e/models/Asset/Image/Thumbnail/Processor.php#L187

**Comment by dkarlovi** [comment]
@brusch it seems this should be done if you've tried cache and failed, not always, no?

**Comment by dvesh3** [comment]
@dkarlovi yes, it should only be done on cache failure. please see https://github.com/pimcore/pimcore/pull/15351

**Comment by dvesh3** [comment]
Fixed by #15351
